src/sentry/tasks/reports/prepare_organization_report.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from django.db.models import F
+
+from sentry import features
+from sentry.models import Organization, OrganizationMember, OrganizationStatus
+from sentry.tasks.base import instrumented_task
+from sentry.tasks.reports import deliver_organization_user_report
+
+
+@instrumented_task(
+    name="sentry.tasks.reports.prepare_organization_report",
+    queue="reports.prepare",
+    max_retries=5,
+    acks_late=True,
+)
+def prepare_organization_report(
+    timestamp: float,
+    duration: float,
+    organization_id: int,
+    user_id: int | None = None,
+    dry_run: bool = False,
+) -> None:
+    from sentry.tasks.reports import backend, logger
+
+    try:
+        organization = Organization.objects.get(
+            status=OrganizationStatus.VISIBLE, id=organization_id
+        )
+    except Organization.DoesNotExist:
+        logger.warning(
+            "reports.organization.missing",
+            extra={
+                "timestamp": timestamp,
+                "duration": duration,
+                "organization_id": organization_id,
+            },
+        )
+        return
+
+    if features.has("organizations:weekly-report-debugging", organization):
```

Comment:
Should not be done in this PR because it's so big, but I don't believe we're using this feature anymore and it should be removed.